src/matUtils/merge.cpp

```diff
@@ -0,0 +1,907 @@
+                std::string x = samples[k];
+                MAT::Node *s = otherMat.get_node(x);
+                auto ancestors = otherMat.rsearch(x, true);
+                std::string curr = finalMat.root->identifier;
+                std::vector<int> indices;
+                /** Parses through the the ancestors of each sample on otherMat and finds 
+                 * Closest consistent node on baseMat 
+                 **/
+                for (int y = 0; y < ancestors.size(); y++)
+                {
+                    m.lock();
+                    if (consistNodes.find(ac, ancestors[y]->identifier) == true)
+                    {
+
+                        anc[y] = ac->second;
+                        indices.push_back(y);
+
+                        ac.release();
+                    }
+                    m.unlock();
+                }
+
+                int min;
+
+                if (indices.size() != 0)
+                {
+                    min = *std::min_element(indices.begin(), indices.end());
+                    curr = anc[min];
+                }
+
+                //Roots bfs at closest consistent node identified in previous loop
+                //Restricts tree search to a smaller subtree
+                auto bfs = finalMat.breadth_first_expansion(curr);
+                placement_input inp;
+                //std::cout<<bfs[0]->mutations.size()<<std::endl;
+
+                std::vector<std::vector<MAT::Mutation> > node_excess_mutations(bfs.size());
+                std::vector<std::vector<MAT::Mutation> > node_imputed_mutations(bfs.size());
+                size_t best_node_num_leaves = 0;
+                int best_set_difference = s->mutations.size() + bfs[0]->mutations.size() + 1;
+                size_t best_j = 0;
+                size_t num_best = 1;
+                bool best_node_has_unique = false;
+                MAT::Node *best_node = bfs[0];
+                std::vector<bool> node_has_unique(bfs.size(), false);
+                std::vector<size_t> best_j_vec;
+                best_j_vec.emplace_back(0);
+                for (int k = 0; k < bfs.size(); k++)
+                {
+                    inp.T = &finalMat;
+                    inp.node = bfs[k];
+                    inp.missing_sample_mutations = &s->mutations;
+                    inp.excess_mutations = &node_excess_mutations[k];
+                    inp.imputed_mutations = &node_imputed_mutations[k];
+
+                    inp.best_node_num_leaves = &best_node_num_leaves;
+                    inp.best_set_difference = &best_set_difference;
+                    inp.best_node = &best_node;
+                    inp.best_j = &best_j;
+                    inp.num_best = &num_best;
+                    inp.j = k;
+                    inp.has_unique = &best_node_has_unique;
+                    inp.best_j_vec = &best_j_vec;
+                    inp.node_has_unique = &(node_has_unique);
+                    placement(inp, false, true);
```

Comment:
Why not just use mapper2_body?

Reply:
I copied and pasted it in case I would have to modify it at all but I can change it so it just uses mapper2_body.

Reply:
Yes, let's not duplicate code.